### PR TITLE
fix: include file names in the message

### PR DESCRIPTION
**Pull request by lyjeileen**
## Changes
- Include file names in the message

Note: The alert is only be shown in the storybook to show the content being sent.

## Before
Screenshot 2024-05-01 at 9 31 10 AM

## After
Screenshot 2024-05-01 at 9 29 17 AM

### CODE REVIEW

File: src/components/input/multimodal/multimodalInput/multimodalInput.tsx [hunk 1]
```diff
@@ -10,17 +10,21 @@ import BaseInput from '../../baseInput/baseInput'
 import Uploader from '../uploader/uploader'
 
 export default function MultimodalInput(props: MultimodalInputProps) {
-  const [fileCount, setFileCount] = useState(0)
+  const [fileList, setFileList] = useState<string[]>([])
```
Comment (Shiti) [inline src/components/input/multimodal/multimodalInput/multimodalInput.tsx:13]:
```suggestion
  const [fileNames, setFileNames] = useState<string[]>([])
```

File: src/components/input/multimodal/multimodalInput/multimodalInput.tsx [hunk 2]
```diff
@@ -10,17 +10,21 @@ import BaseInput from '../../baseInput/baseInput'
...
   const [messageId, setMessageId] = useState(getUUID())
   const [filePreviewsContainer, setFilePreviewsContainer] =
     useState<HTMLDivElement>()
   const [errorMessagesContainer, setErrorMessagesContainer] =
     useState<HTMLDivElement>()
   const inputRef = useRef<HTMLDivElement>(null)
-  const hasAddedFiles = fileCount > 0
+  const hasAddedFiles = fileList.length > 0
```
Comment (Shiti) [inline src/components/input/multimodal/multimodalInput/multimodalInput.tsx:20]:
```suggestion
  const hasAddedFiles = fileNames.length > 0
```

File: src/components/input/multimodal/multimodalInput/multimodalInput.tsx [hunk 3]
```diff
@@ -42,11 +46,12 @@ export default function MultimodalInput(props: MultimodalInputProps) {
     if (hasAddedFiles) {
       formattedMessage.id = messageId
       formattedMessage.format = 'multipart'
+      formattedMessage.data.files = fileList
     }
 
     props.ws.send(formattedMessage)
     setMessageId(getUUID())
-    setFileCount(0)
+    setFileList([])
```
Comment (Shiti) [inline src/components/input/multimodal/multimodalInput/multimodalInput.tsx:54]:
```suggestion
    setFileNames([])
```

File: src/components/input/multimodal/multimodalInput/multimodalInput.tsx [hunk 4]
```diff
@@ -10,17 +10,21 @@ import BaseInput from '../../baseInput/baseInput'
...
     useState<HTMLDivElement>()
   const [errorMessagesContainer, setErrorMessagesContainer] =
     useState<HTMLDivElement>()
   const inputRef = useRef<HTMLDivElement>(null)
-  const hasAddedFiles = fileCount > 0
+  const hasAddedFiles = fileList.length > 0
 
-  function handleFileCountChange(fileCountChange: 1 | -1) {
-    setFileCount((prev) => prev + fileCountChange)
+  function handleFileListChange(action: 'add' | 'remove', fileName: string) {
```
Comment (Shiti) [inline src/components/input/multimodal/multimodalInput/multimodalInput.tsx:22]:
```suggestion
  function handleFileUpdates(action: 'add' | 'remove', fileName: string) {
```

File: src/components/input/multimodal/multimodalInput/multimodalInput.tsx [hunk 5]
```diff
@@ -64,7 +69,7 @@ export default function MultimodalInput(props: MultimodalInputProps) {
             maxFileSize={props.maxFileSize}
             uploadFileEndpoint={props.uploadFileEndpoint}
             deleteFileEndpoint={props.deleteFileEndpoint}
-            handleFileCountChange={handleFileCountChange}
+            handleFileListChange={handleFileListChange}
```
Comment (Shiti) [inline src/components/input/multimodal/multimodalInput/multimodalInput.tsx:72]:
```suggestion
            onFileUpdates={handleFileListChange}
```

File: src/components/input/multimodal/uploader/uploader.tsx [hunk 6]
```diff
@@ -151,22 +153,39 @@ function Uploader(props: UploaderProps) {
     setAddedFiles((prev) => prev.filter((file) => file.id !== id))
   }
 
+  function updateDuplicateFileName(file: File) {
+    let fileName = file.name
+    const extensionIndex = fileName.lastIndexOf('.')
+    const baseName = fileName.substring(0, extensionIndex)
```
Comment (Shiti) [inline src/components/input/multimodal/uploader/uploader.tsx:159]:
```suggestion
```

File: src/components/input/multimodal/uploader/uploader.tsx [hunk 7]
```diff
@@ -151,22 +153,39 @@ function Uploader(props: UploaderProps) {
...
+    const fileNameCount = fileNamesRef.current[fileName]
+
+    if (fileNameCount) {
+      const newCount = fileNameCount + 1
```
Comment (Shiti) [inline src/components/input/multimodal/uploader/uploader.tsx:163]:
```suggestion
      const newCount = fileNameCount + 1
       const extensionIndex = fileName.lastIndexOf('.')
       const baseName = fileName.substring(0, extensionIndex)
```

File: src/components/input/multimodal/uploader/uploader.tsx [hunk 8]
```diff
@@ -151,22 +153,39 @@ function Uploader(props: UploaderProps) {
     setAddedFiles((prev) => prev.filter((file) => file.id !== id))
   }
 
+  function updateDuplicateFileName(file: File) {
```
Comment (Shiti) [inline src/components/input/multimodal/uploader/uploader.tsx:156]:
```suggestion
  function getFileName(file: File) {
```

File: src/components/input/multimodal/uploader/uploader.tsx [hunk 9]
```diff
@@ -188,31 +207,45 @@ function Uploader(props: UploaderProps) {
         handleSuccessfulUpload(response.data, temporaryFileId)
```
Comment (Shiti) [inline src/components/input/multimodal/uploader/uploader.tsx:207]:
```suggestion
        handleSuccessfulUpload(response.data, newAddedFile.id)
```

File: src/components/input/multimodal/uploader/uploader.tsx [hunk 10]
```diff
@@ -188,31 +207,45 @@ function Uploader(props: UploaderProps) {
...
       })
       .catch((error) => {
-        props.handleFileCountChange(-1)
+        props.handleFileListChange('remove', fileName)
+        fileNamesRef.current = {
+          ...fileNamesRef.current,
+          [fileName]: fileNamesRef.current[file.name]--,
+        }
         handleFailedUpload(file.name, temporaryFileId, error.response?.data)
```
Comment (Shiti) [inline src/components/input/multimodal/uploader/uploader.tsx:215]:
```suggestion
        handleFailedUpload(file.name, newAddedFile.id, error.response?.data)
```

File: src/components/input/multimodal/uploader/uploader.tsx [hunk 11]
```diff
@@ -151,22 +153,39 @@ function Uploader(props: UploaderProps) {
...
+
+    return fileName
+  }
+
   function uploadFile(file: File) {
     const formData = new FormData()
     const controller = new AbortController()
     formData.append('file', file)
-
+    const fileName = updateDuplicateFileName(file)
```
Comment (Shiti) [inline src/components/input/multimodal/uploader/uploader.tsx:178]:
```suggestion
    const fileName = getFileName(file)
    const updatedFile = new File([file], fileName, { type: file.type })
    formData.append('file', updatedFile)
```

File: src/components/types.ts [hunk 12]
```diff
@@ -189,8 +189,8 @@ export interface UploaderProps {
   deleteFileEndpoint: string
   /** Used in the API request to link the file with the message that's going to be sent. */
   messageId: string
-  /** A function to handle changes in the file count. Parent component should use this to track file count change and handle submit accordingly. */
-  handleFileCountChange: (fileCountChange: 1 | -1) => void
+  /** A function to handle changes in the file list. Parent component should use this to track file names and handle submit accordingly. */
+  handleFileListChange: (action: 'add' | 'remove', fileName: string) => void
```
Comment (Shiti) [inline src/components/types.ts:193]:
```suggestion
  onFileUpdate: (action: 'add' | 'remove', fileName: string) => void
```

File: src/components/types.ts [hunk 13]
```diff
@@ -205,7 +205,7 @@ export type MultimodalInputProps = TextInputProps &
   Omit<
     UploaderProps,
     | 'messageId'
-    | 'handleFileCountChange'
+    | 'handleFileListChange'
```
Comment (Shiti) [inline src/components/types.ts:208]:
```suggestion
    | 'onFileUpdate'
```

File: src/components/input/multimodal/uploader/uploader.tsx [hunk 14]
```diff
@@ -188,31 +207,45 @@ function Uploader(props: UploaderProps) {
...
       const deleteUrl = `${props.deleteFileEndpoint}?message-id=${props.messageId}&file-id=${file.id}`
+
       axios
         .delete(deleteUrl)
         .then(() => {
           return setAddedFiles((prev) => prev.filter((_, i) => i !== index))
         })
         .catch(() => {
-          props.handleFileCountChange(1)
+          props.handleFileListChange('add', fileName)
```
Comment (Shiti) [inline src/components/input/multimodal/uploader/uploader.tsx:236]:
```suggestion
          props.onFileUpdate('add', fileName)
```

File: src/components/input/multimodal/uploader/uploader.tsx [hunk 15]
```diff
@@ -188,31 +207,45 @@ function Uploader(props: UploaderProps) {
...
 
   function handleDelete(file: FileInfo, index: number) {
     setErrorMessages([])
-    props.handleFileCountChange(-1)
+    const fileName = file.name
+    fileNamesRef.current = {
+      ...fileNamesRef.current,
+      [fileName]: fileNamesRef.current[fileName]--,
+    }
+    props.handleFileListChange('remove', fileName)
```
Comment (Shiti) [inline src/components/input/multimodal/uploader/uploader.tsx:226]:
```suggestion
    props.onFileUpdate('remove', fileName)
```

File: src/components/input/multimodal/uploader/uploader.tsx [hunk 16]
```diff
@@ -188,31 +207,45 @@ function Uploader(props: UploaderProps) {
         handleSuccessfulUpload(response.data, temporaryFileId)
       })
       .catch((error) => {
-        props.handleFileCountChange(-1)
+        props.handleFileListChange('remove', fileName)
```
Comment (Shiti) [inline src/components/input/multimodal/uploader/uploader.tsx:210]:
```suggestion
        props.onFileUpdate('remove', fileName)
```

File: src/components/input/multimodal/uploader/uploader.tsx [hunk 17]
```diff
@@ -151,22 +153,39 @@ function Uploader(props: UploaderProps) {
...
+      name: fileName,
       loadingProgress: 0,
       abortController: controller,
       id: temporaryFileId,
     }
 
     setAddedFiles((prev) => [...prev, newAddedFile])
-
-    props.handleFileCountChange(1)
+    props.handleFileListChange('add', fileName)
```
Comment (Shiti) [inline src/components/input/multimodal/uploader/uploader.tsx:188]:
```suggestion
    props.onFileUpdate('add', fileName)
```

File: src/components/input/multimodal/uploader/uploader.tsx (hunk 9 above)
Comment (Shiti) [reply src/components/input/multimodal/uploader/uploader.tsx:207]: I think references to temporaryFileId in callbacks can be replaced with `newAddedFile.id`

**Review by Shiti — CHANGES_REQUESTED** [review]
I've suggested minor renames to clarify functionality.

File: src/components/input/multimodal/uploader/uploader.tsx (hunk 8 above)
Comment (kaseyvee) [inline src/components/input/multimodal/uploader/uploader.tsx:156]: Would the following be a more accurate name?
```suggestion
  function updateFileNameIfDuplicate(file: File) {
```

File: src/components/types.ts [hunk 18]
```diff
@@ -189,8 +189,8 @@ export interface UploaderProps {
   deleteFileEndpoint: string
   /** Used in the API request to link the file with the message that's going to be sent. */
   messageId: string
-  /** A function to handle changes in the file count. Parent component should use this to track file count change and handle submit accordingly. */
-  handleFileCountChange: (fileCountChange: 1 | -1) => void
+  /** A function to handle changes in the file list. Parent component should use this to track file names and handle submit accordingly. */
```
Comment (kaseyvee) [inline src/components/types.ts:192]:
```suggestion
  /** A function to handle changes in the file list. The parent component should use this to track file names and handle submit accordingly. */
```

**Review by Shiti — APPROVED** [review]

**Review by Shiti — APPROVED** [review]

**Review by Shiti — APPROVED** [review]

**Review by kaseyvee — APPROVED** [review]

**Pull request merged**